### PR TITLE
First Draft for the new documentation website and the new API 

### PR DESCRIPTION
**Pull request by jjfumero**
This PR includes a new style of documentation using sphinx. 

The new documentation is hosted here: https://tornadovm.readthedocs.io/en/feat-documentation/index.html 

This is the first draft of the new documentation and the new style.  Pending things to do after this PR gets merged:

- Update the README file with the new links 
- Merge with the new API branch

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,28 @@
...
-TornadoVM currently has three backends: OpenCL, NVIDIA CUDA PTX and SPIR-V.
+TornadoVM currently has three backends that generate OpenCL C, NVIDIA CUDA PTX, and SPIR-V binary.
 Developers can chose which backends to install and run.
 
 ----------------------
 
 **Website**: [tornadovm.org](https://www.tornadovm.org)
 
-For a quick introduction please read the following [FAQ](tornado-assembly/src/docs/15_FAQ.md).
+For a quick introduction please read the following [FAQ](https://tornadovm.readthedocs.io/en/feat-documentation/faq.html).
```
Comment (stratika) [inline README.md:17]: Will the link work once, we merge the `feat-documentation` branch?

Comment (jjfumero) [reply README.md:17]: yes, it will work. When we do the release, we will point these to the master branch

Comment (jjfumero) [reply README.md:17]: No, we need to change this after we merge.

File: docs/source/faq.rst [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+1. Which programming languages does TornadoVM support?
+------------------------------------------------------
+
+TornadoVM primarily supports Java. However, with the integration with GraalVM you can call your TornadoVM-compatible Java code through other
+programming languages supported by GraalVM's polyglot runtime (e.g., Python, R, Ruby, Javascript, Node.js, etc).
+
+`Here <https://github.com/beehive-lab/TornadoVM/tree/master/tornado-examples/src/main/java/uk/ac/manchester/tornado/examples/polyglot>`__
+you can find examples of how to use TornadoVM with GraalVM Polyglot.
+
+5. Is TornadoVM a DSL?
```
Comment (orionpapadakis) [inline docs/source/faq.rst:59]: Maybe add what the DSL acronym stands for?

File: docs/source/ide-integration.rst [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+Change the IntelliJ maximum memory to 2 GB or more `(instructions) <https://www.jetbrains.com/help/idea/increasing-memory-heap.html#d1366197e127>`__.
+
+For Intellij to pickup the required Tornado dependencies from the poms, go to **View > Tool Windows > Maven** and select **jdk-8, ptx-backend,
+opencl-backend** under profiles.
+
+Required Plugins:
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Open IntelliJ and go to **Preferences > Plugins > Browse Repositories**.
```
Comment (orionpapadakis) [inline docs/source/ide-integration.rst:19]: I think those instructions need to be updated

**Review by orionpapadakis — COMMENTED** [review]
Just 2 minor comments

**Review by stratika — CHANGES_REQUESTED** [review]
One additional comment regards whether we should change the uploaded logo to use one with transparent background. Because this has a frame that breaks the white background.

File: docs/source/index.rst [hunk 4]
```diff
@@ -0,0 +1,48 @@
+TornadoVM
+===================================
+
+.. note::
+
+   This project is under active development.
```
Comment (stratika) [inline docs/source/index.rst:6]: Remove this note.

File: docs/source/ide-integration.rst (hunk 3 above)
Comment (jjfumero) [reply docs/source/ide-integration.rst:19]: Yes. let's do that in a separate PR.

**Comment by jjfumero** [comment]
> One additional comment regards whether we should change the uploaded logo to use one with transparent background. Because this has a frame that breaks the white background.

I updated with the correct logo format

File: README.md [hunk 5]
```diff
@@ -1,30 +1,28 @@
...
-TornadoVM is a plug-in to OpenJDK and GraalVM that allows programmers to automatically run Java programs on
-heterogeneous hardware. TornadoVM currently targets OpenCL-compatible devices and it runs on multi-core CPUs, dedicated
-GPUs (NVIDIA, AMD), integrated GPUs (Intel HD Graphics and ARM Mali), and FPGAs (Intel and Xilinx).
+TornadoVM is a plug-in to OpenJDK and GraalVM that allows programmers to automatically run Java programs on heterogeneous hardware. 
+TornadoVM currently targets OpenCL, PTX and SPIR-V compatible devices and it runs on multi-core CPUs, dedicated
+GPUs (Intel, NVIDIA, AMD), integrated GPUs (Intel HD Graphics and ARM Mali), and FPGAs (Intel and Xilinx).
 
 
-TornadoVM currently has three backends: OpenCL, NVIDIA CUDA PTX and SPIR-V.
+TornadoVM currently has three backends that generate OpenCL C, NVIDIA CUDA PTX, and SPIR-V binary.
```
[the remark this answers is not in the document]
Comment (jjfumero) [reply README.md:10]: I think it is fine the way it is

File: docs/source/publications.rst [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+      title = {{Dynamic Application Reconfiguration on Heterogeneous Hardware.}},
+      booktitle = {Proceedings of the 15th ACM SIGPLAN/SIGOPS International Conference on Virtual Execution Environments},
+      series = {VEE '19},
+      year = {2019},
+      doi = {10.1145/3313808.3313819},
+      publisher = {Association for Computing Machinery}
+   }
+
+
+If you are using Tornado 0.1 (Initial release), please use the following citation in your work.
```
[the remark this answers is not in the document]
Comment (jjfumero) [reply docs/source/publications.rst:23]: Tornado. In 0.1 we had the name `Tornado`. After the dynamic reconfiguration paper, we renamed it to `TornadoVM`.

**Comment by jjfumero** [comment]
Proposed changed done

File: README.md [hunk 7]
```diff
@@ -1,30 +1,28 @@
 # TornadoVM
 
-<img align="right" width="250" height="250" src="etc/tornadoVM_Logo.jpg">
+<img align="left" width="250" height="250" src="etc/tornadoVM_Logo.jpg">
 
-TornadoVM is a plug-in to OpenJDK and GraalVM that allows programmers to automatically run Java programs on
-heterogeneous hardware. TornadoVM currently targets OpenCL-compatible devices and it runs on multi-core CPUs, dedicated
-GPUs (NVIDIA, AMD), integrated GPUs (Intel HD Graphics and ARM Mali), and FPGAs (Intel and Xilinx).
+TornadoVM is a plug-in to OpenJDK and GraalVM that allows programmers to automatically run Java programs on heterogeneous hardware. 
+TornadoVM targets OpenCL, PTX and SPIR-V compatible devices which include on multi-core CPUs, dedicated
```
Comment (stratika) [inline README.md:6]: remove `on`

**Review by stratika — COMMENTED** [review]
The `index.rst` file has not changed.

**Comment by jjfumero** [comment]
Done

**Comment by stratika** [comment]
@jjfumero some files have not changed. I see the previous text in it. Have you pushed all changes?

**Comment by jjfumero** [comment]
for the rendered website, it needs a few minutes. Also, I suggest reviewing changes in a new browser instance, to clean the cache.

**Comment by stratika** [comment]
> for the rendered website, it needs a few minutes. Also, I suggest reviewing changes in a new browser instance, to clean the cache.

I see the changes in GitHub. I will refresh in case it has not fetched the changes.

File: docs/source/index.rst [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  :width: 100
+  :alt: Sample Text
+  :align: left
+
+TornadoVM is a plug-in to OpenJDK and GraalVM that allows developers to automatically run Java programs on heterogeneous hardware. 
+TornadoVM currently targets OpenCL-compatible, PTX-compatible and Level-Zero compatible devices which include multi-core CPUs,
+dedicated GPUs (NVIDIA, AMD), integrated GPUs (Intel HD Graphics and ARM Mali), and FPGAs (Intel and Xilinx).
+
+TornadoVM currently has three backends: OpenCL, NVIDIA CUDA PTX, and SPIR-V.
+Developers can chose which backends to install and run.
```
Comment (stratika) [inline docs/source/index.rst:14]: chose -> choose

File: docs/source/introduction.rst [hunk 9]
```diff
@@ -0,0 +1,47 @@
+Introduction to TornadoVM
+=========================
+
+
+.. image:: images/logo.png
+  :width: 100
+  :alt: Sample Text
+  :align: left
+
+TornadoVM is a plug-in to OpenJDK and GraalVM that allows developers to automatically run Java programs on heterogeneous hardware.
```
[the remark this answers is not in the document]
Comment (jjfumero) [reply docs/source/introduction.rst:10]: OpeJDK-based distributions.

**Review by mairooni — APPROVED** [review]

**Review by orionpapadakis — APPROVED** [review]

**Pull request merged**